build.sbt

```diff
@@ -156,15 +156,21 @@ lazy val core = crossProject(JSPlatform, JVMPlatform)
   .jvmSettings(
     libraryDependencies ++= Seq(
       "org.typelevel" %% "cats-kernel" % "2.6.1"
-    ) ++ specs
+    ) ++ specs ++ (CrossVersion.partialVersion(scalaVersion.value) match {
+      case Some((3, _)) => Nil
```

Comment:
Wasn't collection compat removed?

Reply:
it is a dependency of saddle now on 2.13, mima complains. I think we can put the dependency back (I did not revert the source changes). 

I am facing an other problem, on 2.13 the `linalg` module depends on `core` and on the inlined macro module. I don't know how to convince sbt to make this conditional on the language version

Reply:
In an precedent commit we removed that dependency. I don't understand why we reintroduce it. This PR drops support for scala2.12, so `collection-compat` shouldn't be needed anymore.

Reply:
It is not needed, but saddle_2.13 version 3.3.0 depends on it in its pom.xml . Mima complains that the dependency is missing.  otherwise this PR compiles fine if we exclude the collection-compat.